lib/floe/workflow/context.rb

```diff
@@ -9,9 +9,7 @@ class Context
       # @param input [Hash] (default: {})
       def initialize(context = nil, input: nil, credentials: {})
         context = JSON.parse(context) if context.kind_of?(String)
-
-        input ||= {}
-        input = JSON.parse(input) if input.kind_of?(String)
+        input   = JSON.parse(input || "{}")
```

Comment:
Like how the `if input...` has been removed.

Would prefer to change the default `input: "{}"` and drop the `input || "{}"`

Reply:
the `input ||= {}` was if someone explicitly passed `:input => nil` it would still default to `{}`

Reply:
Heh. Yea.

I guess in general I want to keep the data closer to what the user provided.

Reply:
I think the better question is "what does step functions do?", if I do
```
aws stepfunctions start-execution --input "null"
```

I get "null" output
```
{
    "executionArn": "arn:aws:states:us-east-1:123456789012:execution:Succeed:6cae58d1-1aa6-415f-a010-ddc730d6b81f",
    "stateMachineArn": "arn:aws:states:us-east-1:123456789012:stateMachine:Succeed",
    "name": "6cae58d1-1aa6-415f-a010-ddc730d6b81f",
    "status": "SUCCEEDED",
    "startDate": "2024-06-24T14:50:47.849000-04:00",
    "stopDate": "2024-06-24T14:50:47.876000-04:00",
    "input": "null",
    "inputDetails": {
        "included": true
    },
    "output": "null",
    "outputDetails": {
        "included": true
    }
}
```

So I think actually `{}` if not provided, but `nil` if explicitly passed `"null"` is actually correct.

Reply:
I think we are in total agreement here.

(I was just squabbling over who converted a `nil => {} / "null" => nil`, and was thinking it was the cli. But not my hill)

Reply:
> (I was just squabbling over who converted a nil => {} / "null" => nil, and was thinking it was the cli. But not my hill)

Hey I'm open to why we shouldn't do it Context, but if we did it in cli then the same logic wouldn't apply to anyone using Context/Workflow as ruby classes right?

Reply:
> Hey I'm open to why we shouldn't do it Context, but if we did it in cli then the same logic wouldn't apply to anyone using Context/Workflow as ruby classes right?

Totally.
I double talk here. I feel that workflow is the public interface, but I want people to pass in a context.

Maybe we need a public interface for creating a context, like we have `Workflow#load`

Ok, the only thing I care about is when you are creating a Map, do you have a context for each parallel branch? Are you able to handle this conversion?

If so, lets :shipit:

Reply:
> Ok, the only thing I care about is when you are creating a Map, do you have a context for each parallel branch? Are you able to handle this conversion?

Yes, each `ItemProcessor` iteration has its own `ItemProcessorContext` which lives under the top-level `Context["State"]` for the Map state.